hail/python/hailtop/hailctl/batch/batch_cli_utils.py

```diff
@@ -35,32 +36,32 @@ def bool_string_to_bool(bool_string):
     raise ValueError("Input could not be resolved to a bool")
 
 
-def separated_format(lod: LoD, delim: str) -> None:
-    writer = csv.writer(sys.stdout, delimiter=delim)
-    # construct header dictionary from keys of lod
-    header = {k: k for (k, v) in lod[0].items()}
+def separated_format(table_data: TableData, delim: str) -> str:
+    output = io.StringIO()
+    writer = csv.writer(output, delimiter=delim)
+    header = {k: k for k in table_data[0].keys()}
```

Comment:
This can just be `header = table_data[0].keys()`. Let's also place `assert len(table_data) > 0` just before this line to convey that access should be valid.